### PR TITLE
Hide Window extension methods by default and add helper Window.js_backtrace method

### LINKED ISSUE #26858: Add API for dumping current JS/Rust backtraces
This will make it easier to figure out the callers of certain JS code when investigating web compat issues.

### PR DESCRIPTION
**Pull request by jdm**
Fixes #26858.

### CODE REVIEW

File: components/script/Cargo.toml [hunk 1]
```diff
@@ -33,7 +33,7 @@ serde_json = "1.0"
 accountable-refcell = { version = "0.2.0", optional = true }
 app_units = "0.7"
 arrayvec = "0.5.1"
-backtrace = { version = "0.3", optional = true }
+backtrace = { version = "0.3" }
```
Comment (atouchet) [inline components/script/Cargo.toml:36]:
```suggestion
backtrace = "0.3"
```

**Review by Manishearth — APPROVED** [review]
Probably should mention the js bt thing in the PR title

**Pull request merged**